src/main/java/htsjdk/samtools/CSIIndex.java

```diff
@@ -0,0 +1,466 @@
+package htsjdk.samtools;
+
+import htsjdk.samtools.seekablestream.SeekableStream;
+import htsjdk.samtools.util.RuntimeIOException;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.*;
+
+
+public class CSIIndex extends AbstractBAMFileIndex implements BrowseableBAMIndex {
```

Comment:
Could you add class javadoc that explains what the CSI index is and gives some intuition about how it's different from the Bai index.